pkg/sysfs/system.go

```diff
@@ -995,3 +1018,20 @@ func (sys *system) discoverCache(path string) error {
 
 	return nil
 }
+
+var eppNames = func() map[string]EPP {
+	n := make(map[string]EPP, EPPUnknown-1)
+	for i := 0; i < len(_EPP_index); i++ {
+		n[_EPP_name[_EPP_index[i]:_EPP_index[i+1]]] = EPP(i)
+	}
+
+	return n
+}()
+
+// EPPFromString converts string to EEP value
+func EPPFromString(s string) EPP {
+	if e, ok := eppNames[s]; ok {
+		return e
+	}
+	return EPPUnknown
+}
```

Comment:
I have to say that I'm genuinely surprised that with that much boilerplate and a full blown generator, stringer is unable to generate this direction at all.

Reply:
Yeah, I think that is a very unfortunate major omission from the tool. Maybe this is an overkill for this 5-value enum. Otoh, I tried to explore something nice and reliable that could possibly be used across cri-rm.

I think I'll let this proof for a while, at least sleep over it...

Reply:
> Yeah, I think that is a very unfortunate major omission from the tool. Maybe this is an overkill for this 5-value enum. Otoh, I tried to explore something nice and reliable that could possibly be used across cri-rm.

I do agree that's a worthy goal. It's just that ideally

  - the generator should be able to generate code in both the enum-to-string and string-to-enum directions
  - we'd have dependency rules in place to only invoke the generator when necessary

Working dependency rules might be just too much of a hassle/require too much manual Makefile boilerplate to do with the available go tooling. In that case a perfectly acceptable alternative would be to not commit the generated enum-related go source to the repo but instead always generate it. Then the Makefile-related misery would be reduced to simply having the necessary rules in place for every go source to generate the corresponding stringer-go source files unconditionally.

> I think I'll let this proof for a while, at least sleep over it...

Fair enough.

Reply:
Dropped stringer and returned to my original implementation, with the improvement you suggested. I don't consider this the most beautiful but don't know how to make it neater, either 😬

Reply:
All things considered, I think this is probably the best compromise of all the things we tried.